docs/frontend/in-depth/autosave.md

```diff
@@ -0,0 +1,154 @@
+# Autosave
+
+Autosave is the way of saving objects in system. Main idea about that is to prevent user from leaving any page without saved data and to make sure made changes are conscious. The idea is build at top of:
+- [**Scoping data**](#scoping-data)
+- [**Handling validation errors**](#handling-validation-errors)
+- [**Saving data per context at which user have focus**](#saving-data-per-context-at-which-user-have-focus)
+- [**Giving feedback**](#giving-feedback)
+- [**Reminding about errors and not saved data**](#reminding-about-errors-and-not-saved-data)
+
+### Scoping data
+---
+
+To give user proper feedback for current context data have to be scoped by the unique identifier. Process of generating identifiers is same for each context. Converted Component name to Camel Case `MyComponent => myComponent` is treated as a scope of user feedback.
```

Comment:
To give a user
for a given context, data has to be scoped
identifiers is the same for every context